logging/logging.go

```diff
@@ -24,77 +24,253 @@ package logging
+	l.log(WARNING, &format, args...)
+}
+
+// Notice logs a message using NOTICE as log level.
+func (l *Logger) Notice(args ...interface{}) {
+	l.log(NOTICE, nil, args...)
+}
+
+// Noticef logs a message using NOTICE as log level.
+func (l *Logger) Noticef(format string, args ...interface{}) {
+	l.log(NOTICE, &format, args...)
+}
+
+// Info logs a message using INFO as log level.
+func (l *Logger) Info(args ...interface{}) {
+	l.log(INFO, nil, args...)
+}
+
+// Infof logs a message using INFO as log level.
+func (l *Logger) Infof(format string, args ...interface{}) {
+	l.log(INFO, &format, args...)
+}
+
+// Debug logs a message using DEBUG as log level.
+func (l *Logger) Debug(args ...interface{}) {
+	l.log(DEBUG, nil, args...)
+}
+
+// Debugf logs a message using DEBUG as log level.
+func (l *Logger) Debugf(format string, args ...interface{}) {
+	l.log(DEBUG, &format, args...)
+}
+
+func NewEncoderConfig() zapcore.EncoderConfig {
+	return zapcore.EncoderConfig{
+		// Keys can be anything except the empty string.
+		TimeKey:        "ts",
+		LevelKey:       "level",
+		NameKey:        "name",
+		CallerKey:      "caller",
+		MessageKey:     "message",
+		StacktraceKey:  "stacktrace",
+		LineEnding:     zapcore.DefaultLineEnding,
+		EncodeLevel:    zapcore.CapitalLevelEncoder,
+		EncodeTime:     zapcore.ISO8601TimeEncoder,
+		EncodeDuration: zapcore.StringDurationEncoder,
+		EncodeCaller:   zapcore.ShortCallerEncoder,
+	}
+}
+
+func NewConfig() zap.Config {
+	return zap.Config{
+		Level:            zap.NewAtomicLevelAt(zap.DebugLevel),
+		Development:      false,
+		Encoding:         "console",
+		EncoderConfig:    NewEncoderConfig(),
+		OutputPaths:      []string{"stderr"},
+		ErrorOutputPaths: []string{"stderr"},
+	}
+}
+
+func initLogger() (err error) {
+	cfg := config.GetConfig()
+	backendLevel := getZapLevel(cfg.GetString("logging.level"))
+	consoleEncoder := zapcore.NewConsoleEncoder(NewEncoderConfig())
+	jsonEncoder := zapcore.NewJSONEncoder(NewEncoderConfig())
+	msgPriority := zap.LevelEnablerFunc(func(lvl zapcore.Level) bool {
+		return lvl >= backendLevel
+	})
+
+	var backends []zapcore.Core
 	for _, name := range cfg.GetStringSlice("logging.backends") {
 		switch name {
 		case "file":
 			filename := cfg.GetString("logging.file.path")
 			file, err := os.Create(filename)
 			if err != nil {
-				return nil, err
+				return err
 			}
-			backend = logging.NewLogBackend(file, "", 0)
+			backends = append(backends, zapcore.NewCore(consoleEncoder, zapcore.Lock(file), msgPriority))
 		case "syslog":
-			backend, err = logging.NewSyslogBackend("")
+			w, err := syslog.New(syslog.LOG_CRIT, "")
 			if err != nil {
-				return nil, err
+				return err
 			}
+			backends = append(backends, zapcore.NewCore(consoleEncoder, zapcore.AddSync(w), msgPriority))
 		case "stderr":
-			backend = logging.NewLogBackend(os.Stderr, "", 0)
+			backends = append(backends, zapcore.NewCore(consoleEncoder, zapcore.Lock(os.Stderr), msgPriority))
+		case "json":
```

Comment:
this is a bit weird as json log only to stdout. It seems that json should be an encoder ? maybe you can add a section to the logging one like:

logging:
  encoder: json

so that we can use json for all the backends